examplesV4/tokenCreationAndPaymentExample.php

```diff
@@ -0,0 +1,221 @@
+ * Full params available in the documentation
+ */
+
+$order->withBackRef('http://path/to/your/returnUrlScript')
+    ->withOrderRef('MerchantOrderRef')
+    ->withCurrency('RON')
+    ->withOrderDate(gmdate('Y-m-d\TH:i:sP'))
+    ->withOrderTimeout(1000)
+    ->withPayMethod('CCVISAMC');
+
+/**
+ * Create new product
+ */
+$product = new Product();
+
+/**
+ * Setup the product params
+ *
+ * Full params available in the documentation
+ */
+$product->withCode('PCODE01')
+    ->withName('PNAME01')
+    ->withPrice(100.0)
+    ->withVAT(24.0)
+    ->withQuantity(1);
+
+/**
+ * Add the product to the order
+ */
+$order->addProduct($product);
+
+/**
+ * Create another product
+ */
+$product = new Product();
+
+/**
+ * Setup the product params
+ *
+ * Full params available in the documentation
+ */
+$product->withCode('PCODE02')
+    ->withName('PNAME02')
+    ->withPrice(200.0)
+    ->withVAT(24.0)
+    ->withQuantity(1);
+
+/**
+ * Add the second product to the same order
+ */
+$order->addProduct($product);
+
+/**
+ * Create new billing address
+ */
+$billing = new Billing();
+
+/**
+ * Setup the billing address params
+ *
+ * Full params available in the documentation
+ */
+$billing->withAddressLine1('Address1')
+    ->withAddressLine2('Address2')
+    ->withCity('City')
+    ->withCountryCode('RO')
+    ->withEmail('john.doe@mail.com')
+    ->withFirstName('FirstName')
+    ->withLastName('LastName')
+    ->withPhoneNumber('40123456789')
+    ->withIdentityCardNumber('111222');
+
+/**
+ * Create new delivery address
+ *
+ * If you want to have the same delivery as billing, skip these two steps
+ * and pass the Billing $billing object to the request twice
+ */
+$delivery = new Delivery();
+
+/**
+ * Setup the delivery address params
+ *
+ * Full params available in the documentation
+ */
+$delivery->withAddressLine1('Address1')
+    ->withAddressLine2('Address2')
+    ->withCity('City')
+    ->withCountryCode('RO')
+    ->withEmail('john.doe@mail.com')
+    ->withFirstName('FirstName')
+    ->withLastName('LastName')
+    ->withPhoneNumber('40123456789');
+
+/**
+ * Create new Card with params:
+ *
+ * Credit Card Number
+ * Credit Card Expiration Month
+ * Credit Card Expiration Year
+ * Credit Card CVV (Security Code)
+ * Credit Card Owner
+ */
+$card = new Card(
+    '4111111111111111',
+    '01',
+    date("Y", strtotime("+1 year")),
+    123,
+    'Card Owner Name'
+);
+
+/**
+ * tokenize card for further token payments
+ */
+$card->enableTokenCreation();
+
+/**
+ * Create new Request with params:
+ *
+ * Config object
+ * Order object
+ * Billing object
+ * Delivery (or Billing object again, if you want to have the delivery address the same as the billing address)
+ * User object
+ */
+$request = new Request($cfg, $order, $billing, $delivery, $user, PaymentsV4::API_VERSION_V4);
+$requestWithToken = new Request($cfg, $order, $billing, $delivery, $user, PaymentsV4::API_VERSION_V4);
+$requestWithToken->getOrder()->withOrderRef('TokenOrderRef');
```

Comment:
requestWithToken should be defined as close to the moment it will be used,  especially when there are cases when it can't be be used. So it could be moved after $cardToken = new CardToken($response->getTokenHash());

Reply:
done